### PR TITLE
Adds an IndexerConnector mode to only sync the local DB

### LINKED ISSUE #25799: Vulnerability scanner: The inventory synchronization events won't be processed if the Indexer module is disabled
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.8.x - 4.9.x | Vulnerability scanner| Manager | Packages/Sources | Any OS |

## Description

It was reported by the users that in some cases, the Wazuh Indexer shows vulnerabilities for packages that are no longer installed
- https://github.com/wazuh/wazuh/issues/24444
- https://github.com/wazuh/wazuh/issues/25663
- https://github.com/wazuh/wazuh/issues/24131

Analyzing the code, it was determined that this might happen when, but not only, the Indexer module hasn't been initialized (disabled by configuration for example). The events from the inventory DB will be dropped in that case causing these DBs to lose the synchronization and leaving dangling events indexed 

https://github.com/wazuh/wazuh/blob/c1ffaafad4551af00ef70fe4356c4093ae1165c5/src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/resultIndexer.hpp#L52

## DoD

The solution to this situation is to change the behaviour of the `indexerConnector` initialization. If it is disabled, instead of not creating the instance we need to create it anyway but prevent it from talking with the indexer. Then, the only operation available at this state is to process the events received from the inventory DB:
- When disabled, the `m_dispatcher` queue will consider all the events as `no-index`. This will update `m_db` but won't impact the index.
https://github.com/wazuh/wazuh/blob/aeeb3c7249b14eec3b7d1e0f40ead99abada7519/src/shared_modules/indexer_connector/src/indexerConnector.cpp#L455-L456
- The `m_syncQueue` shouldn't work when disabled
https://github.com/wazuh/wazuh/blob/aeeb3c7249b14eec3b7d1e0f40ead99abada7519/src/shared_modules/indexer_connector/src/indexerConnector.cpp#L496-L497
- The `m_initializeThread` shouldn't work when disabled
https://github.com/wazuh/wazuh/blob/aeeb3c7249b14eec3b7d1e0f40ead99abada7519/src/shared_modules/indexer_connector/src/indexerConnector.cpp#L521-L522

Tasks
- [x] Update the `indexerConnector` class with the new behavior
- [x] Confirm with manual tests that the synchronization issue can't occur 
- [x] Component tests
- [x] CI Pass

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|Closes #25799|

## Description

This PR adds another constructor to the `IndexerConnector` class to support a simplified behavior that only keeps the local DB synced.
Then, when the module is enabled again it should re-sync all the elements.

Consider that this approach doesn't solve the pre-existent issue of the effects of the configuration of other modules over vulnerability detector, for example: the change of the cluster name could impact in the name of the DB generated, even if the indexer module is disabled or the cluster is disabled

## Tests

To test the feature, I configured a cluster with a worker and a master, and connected it to an Indexer instance.
Then, an agent with vulnerable packages reports to the worker and its CVEs are indexed

After this, the nodes are stopped, the indexer module is disabled and the nodes are restarted.
The agent is upgraded, and the new package is processed but the changes aren't reflected in the index.
Finally, the indexer module is enabled again and the nodes restarted. In the next **syscollector** sync of the agent, a "diff" is triggered and the CVEs are removed from the index

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation

- [x] Added unit tests (for new features)

### CODE REVIEW

**Review by sebasfalcone — COMMENTED** [review]
# Testing

## Indexer disabled - Index status enabled 🟢  
```xml
  <vulnerability-detection>
    <enabled>yes</enabled>
    <index-status>yes</index-status>
    <feed-update-interval>60m</feed-update-interval>
  </vulnerability-detection>

  <indexer>
    <enabled>no</enabled>
    <hosts>
      <host>https://127.0.0.1:9200</host>
    </hosts>
    <ssl>
      <certificate_authorities>
        <ca>/etc/filebeat/certs/root-ca.pem</ca>
      </certificate_authorities>
      <certificate>/etc/filebeat/certs/filebeat.pem</certificate>
      <key>/etc/filebeat/certs/filebeat-key.pem</key>
    </ssl>
  </indexer>
```

VD won't start
```
2024/10/04 14:48:25 wazuh-modulesd:vulnerability-scanner[241859] vulnerabilityScannerFacade.cpp:511 at start(): ERROR: VulnerabilityScannerFacade::start: Indexer cannot be disabled while vulnerability detection index is enabled..
```

It has to do with this:
https://github.com/wazuh/wazuh/blob/bdeb8da465e3bd82ec9529420e8d9cbf2effcc93/src/wazuh_modules/vulnerability_scanner/src/policyManager/policyManager.hpp#L277-L283

Should this be taken into consideration?

## Indexer enabled - Index status disabled
```xml
  <vulnerability-detection>
    <enabled>yes</enabled>
    <index-status>no</index-status>
    <feed-update-interval>60m</feed-update-interval>
  </vulnerability-detection>

  <indexer>
    <enabled>yes</enabled>
    <hosts>
      <host>https://127.0.0.1:9200</host>
    </hosts>
    <ssl>
      <certificate_authorities>
        <ca>/etc/filebeat/certs/root-ca.pem</ca>
      </certificate_authorities>
      <certificate>/etc/filebeat/certs/filebeat.pem</certificate>
      <key>/etc/filebeat/certs/filebeat-key.pem</key>
    </ssl>
  </indexer>
```

Index wasn't created 🟢 
```json
{
  "error": {
    "root_cause": [
      {
        "type": "index_not_found_exception",
        "reason": "no such index [wazuh-states-vulnerabilities-thinkpad-sebas]",
        "index": "wazuh-states-vulnerabilities-thinkpad-sebas",
        "resource.id": "wazuh-states-vulnerabilities-thinkpad-sebas",
        "resource.type": "index_or_alias",
        "index_uuid": "_na_"
      }
    ],
    "type": "index_not_found_exception",
    "reason": "no such index [wazuh-states-vulnerabilities-thinkpad-sebas]",
    "index": "wazuh-states-vulnerabilities-thinkpad-sebas",
    "resource.id": "wazuh-states-vulnerabilities-thinkpad-sebas",
    "resource.type": "index_or_alias",
    "index_uuid": "_na_"
  },
  "status": 404
}
```

Inventory was updated:
```
000_08b2644ff887bc2b6e7f2771a95bc7919c57f7da ==> CVE-2022-40896
000_107521d7667a466d9fad93bd3b72b28eb1a4d95e ==> CVE-2024-6345
000_1286064fae29bcf5cbd5c069469a30ecd80acb73 ==> CVE-2024-5569
000_16cffeffb1fb0bf1b79469741c2e22e9bcbc70f3 ==> CVE-2024-34064,CVE-2024-22195
000_1eb21d2385dfd1574d1e96bc29aeee22c0cfcadc ==> CVE-2023-5752
000_21edef1d387823a53819eddf690ca4fdd64b9b02 ==> CVE-2023-52323
000_26589b3ad7c7e5155cfcd32016fffe8fd4b2ee8d ==> CVE-2024-35195
000_2f6e6738cd66732cba19df00662691fa0f272237 ==> CVE-2024-34064
000_304b3fa5a67fabc52d13b6174b896617ee569704 ==> CVE-2023-28370
000_353254d125f7f147ad53cbb5852d0067c6b0b041 ==> CVE-2021-34141
000_3e263a1db5df71a918c9113db5e88d881541e6b0 ==> CVE-2024-6345
000_40eb34abfd8643da1711cb8f9cb26c83c3cd1d08 ==> CVE-2023-52323
000_448764256dbb038565c2d4ac6b677d34d0dccee5 ==> CVE-2023-25399
000_4687cb839aca0f56a725f900a61b6164f9ee4cac ==> CVE-2024-37891,CVE-2023-45803,CVE-2023-43804
000_4cd14a9aa8a81bdc06fa81c5863cfea9aed9fb95 ==> CVE-2024-0727,CVE-2023-49083,CVE-2023-50782,CVE-2023-23931,CVE-2023-0286
000_4f2f414ea12ef01d85b9dd7c024f647c4afc4b8e ==> CVE-2024-34069
000_557fe453337a6258db3db00fe54f312c0ef64675 ==> CVE-2024-35195,CVE-2023-32681
000_5e02030893451bf11df089758262edf46f293f0c ==> CVE-2024-37891
000_7455fb20b3fb040004ef7eb4f324291041b09bd8 ==> CVE-2024-8775,CVE-2024-0690,CVE-2023-5764
000_7894e948c917e30422dfbd016066c03919f0bfa1 ==> CVE-2023-33733
000_7a35466aa1614afefc9f8e1714aab2146f9229f2 ==> CVE-2023-26112
000_7ad54e5be6321fdd2580d0e2c94514d4ac3b9274 ==> CVE-2024-39689
000_7c9363723fb9c47051af3f91bb4bdd5fe17bb6c9 ==> CVE-2024-0727,CVE-2023-49083,CVE-2023-50782,CVE-2023-23931,CVE-2023-0286
000_7cb61bc171bbebda9bd22488dc7f9cba479bd4b8 ==> CVE-2024-34069
000_839e9ff0a29693a0e226539387552972b1171039 ==> CVE-2023-33953,CVE-2023-32732,CVE-2023-32731,CVE-2023-1428
000_868214fc499fe93a10723467225025300a38705a ==> CVE-2023-29483
000_9283885a528c54eb6bcc32cbeac65bcd85666797 ==> CVE-2024-42353
000_9db8cc5661f2ae3df7e817502f0686cda78250ff ==> CVE-2024-0727,CVE-2023-49083,CVE-2023-50782,CVE-2023-23931,CVE-2023-0286
000_a05fb1b83fbd31b96f68c800facd4e1d5c3b163a ==> CVE-2024-3651
000_a5b6aeb179cbb200a0953fac6238b24d652c37d4 ==> CVE-2023-45139
000_b91b1006184d6bfc4d8f753b3a23fd3f0727615d ==> CVE-2022-29217
000_c1d281d6a87c46f7c19e4e2f6ad594a5a14f89bd ==> CVE-2024-6345
000_c642e925c59f2c99c1aa3b49c643341a4793bc52 ==> CVE-2022-40898
000_d34ec9ed8096bbe1d14289a99a658d3daa8e0ddc ==> CVE-2021-29063
000_e0b134f74758ced25faa0404ef4ba961b98d0613 ==> CVE-2024-39689
000_e0bef742ed057bdee707db3f28f98def1decdaac ==> CVE-2024-35195
000_e5fe32292452f2e05d931a54789992a8a7735af8 ==> CVE-2023-52323
000_e71ab1deea9a482edbff20f4721fc3f45a6a553e ==> CVE-2024-8775,CVE-2024-0690,CVE-2023-5764
000_f862ea4e955d202b0a5aa16e83d96e4ad9cdee09 ==> CVE-2022-40023
000_fdf980e9f8309d6c1c31dc5d6ba273ea81302527 ==> CVE-2022-36087
```

## Enable index status 🟢 
```xml
  <vulnerability-detection>
    <enabled>yes</enabled>
    <index-status>yes</index-status>
    <feed-update-interval>60m</feed-update-interval>
  </vulnerability-detection>

  <indexer>
    <enabled>yes</enabled>
    <hosts>
      <host>https://127.0.0.1:9200</host>
    </hosts>
    <ssl>
      <certificate_authorities>
        <ca>/etc/filebeat/certs/root-ca.pem</ca>
      </certificate_authorities>
      <certificate>/etc/filebeat/certs/filebeat.pem</certificate>
      <key>/etc/filebeat/certs/filebeat-key.pem</key>
    </ssl>
  </indexer>
```

All 63 CVEs from the inventory were published
```json
{
  "count": 63,
  "_shards": {
    "total": 1,
    "successful": 1,
    "skipped": 0,
    "failed": 0
  }
}
```

**Comment by pereyra-m** [comment]
Hi @sebasfalcone 

> Should this be taken into consideration?

Currently we have two ways to disable the indexation, from the module or from the module itself. It'll have more sense in the future when more than one module uses the Indexer Connector.
That error is telling the user that if he expects the vulnerabilities index to work, the module must be enabled also.

**Comment by pereyra-m** [comment]
## Manual tests

We validated the changes with @sebasfalcone.
A Wazuh Indexer was deployed with docker and the development branch was compiled and connected to the indexer.

It wasn't possible to use the already generated VD content for 4.10.x because the base branch of this PR (4.10.1) still doesn't have the most recent changes of the scanner and it fails to complete the scan (https://github.com/wazuh/wazuh/pull/25681).

An Ubuntu Jammy Docker agent was used for the vulnerabilities test.

**Index disabled**: with this option disabled, we generated the content from the snapshot
**Agent installation**: the agent was installed and scanned. The CVEs were reported in the logs and are present in the local DB, but weren't indexed as expected

```
root@fb659c773250:/var/ossec# grep "Processing and publish key" logs/ossec.log  | wc -l
182
```

<details><summary>Details</summary>
<p>

```
root@fb659c773250:/workspaces/wazuh.worktrees/4.10.0# ./src/build/wazuh_modules/vulnerability_scanner/testtool/rocksDBQuery/rocks_db_query_testtool -d /var/ossec/queue/vd/inventory/ -c package
001_022d75f04e03b48f59841a7a2cbabea0d6d32210 ==> CVE-2024-6119,CVE-2024-4603,CVE-2024-41996,CVE-2024-2511,CVE-2024-5535,CVE-2022-40735
001_02b8db7c360707c5978cfecdbdb7e7293cd8ac16 ==> CVE-2024-22365,CVE-2022-28321
001_03fe0e19ee196b1728b85a4dac8faf14948603b8 ==> CVE-2023-29491,CVE-2022-29458,CVE-2019-17595,CVE-2019-17594
001_0739e3ae9cf2dd14ca9a9cad5badd7c6171cfc9a ==> CVE-2022-4415,CVE-2022-3821
001_09b119a1fd53f27516b52bfa5ebc2c9928812b58 ==> CVE-2023-29491,CVE-2022-29458,CVE-2019-17595,CVE-2019-17594
001_0f25e30abb45811cba71c1dd3b17559e660dabe9 ==> CVE-2024-37371,CVE-2024-37370,CVE-2023-36054,CVE-2022-42898
001_0f55e7f6e28422129f2b68d507de34e1188d0718 ==> CVE-2024-28085
001_10fd163869ab59c6ae42ac69961e1ff4313b29b6 ==> CVE-2024-8096,CVE-2024-7264,CVE-2024-2398
001_117940a830aac21aaa039ceae97fbc5ed788c5b4 ==> CVE-2024-22365,CVE-2022-28321
001_125e45f7cf0fdae401da37ee5528cc1d5afb1d0d ==> CVE-2022-4415,CVE-2022-3821
001_20b2a90e2595e6c499f952be5eee29d38f8bedfb ==> CVE-2024-22365,CVE-2022-28321
001_29bc8e8cff2491abf99c48e8d92ef77b9da77b7b ==> CVE-2023-4641,CVE-2013-4235
001_3349b7be200454f0e6d0af3f25ee2b503f21d681 ==> CVE-2024-28835,CVE-2024-28834,CVE-2024-0567,CVE-2024-0553,CVE-2023-5981,CVE-2023-0361
001_35c410fdec0a7818c057483c3ebeb6621085ecdc ==> CVE-2024-28085
001_3b2899e2f2ee513f7d84f1558d407f64addf518c ==> CVE-2024-37371,CVE-2024-37370,CVE-2023-36054,CVE-2022-42898
001_41ba114f43359932416b8e30167017e9da7282c5 ==> CVE-2024-6119,CVE-2023-3817,CVE-2023-1255,CVE-2023-0466,CVE-2023-6129,CVE-2023-0464,CVE-2022-3358,CVE-2023-2650,CVE-2022-3602,CVE-2023-0215,CVE-2022-3786,CVE-2024-5535,CVE-2024-4603,CVE-2024-2511,CVE-2023-5363,CVE-2023-0286,CVE-2023-5678,CVE-2023-3446,CVE-2022-3996,CVE-2023-0216,CVE-2022-40735,CVE-2023-2975,CVE-2023-0401,CVE-2022-4203,CVE-2024-0727,CVE-2022-4304,CVE-2023-6237,CVE-2022-4450,CVE-2023-0217
001_4473f7e794b6a071e75f4b774c74b47874198b5f ==> CVE-2023-2603,CVE-2023-2602
001_4c95954c6c54e78b0fe5417ea65717a3e4b8ec37 ==> CVE-2023-4016
001_4d4102b4c947d9f1e78f24603e5f6b355bd6c28d ==> CVE-2023-29491,CVE-2022-29458,CVE-2019-17595,CVE-2019-17594
001_50f7946dce4417b9f0ac1b5d56c78a8743202b54 ==> CVE-2022-1587,CVE-2022-1586
001_5b1fcc1339bb5a2b99be6041c7e8eea885758366 ==> CVE-2024-28085
001_5f96ac012c8d8df2d431b9dc36b0852c4c6ad627 ==> CVE-2023-39804,CVE-2022-48303
001_607b974e2382ba048fef85df626b7ed8d325152b ==> CVE-2024-28085
001_68d497ecd10c148f8701eaad46ec7bb56b1f328f ==> CVE-2024-38428,CVE-2021-31879
001_8a52a614cde3cc7dbada50cf85990e41ca036164 ==> CVE-2024-28182
001_8b4fb395957df82c5ebc19e583ca211bfa366732 ==> CVE-2024-37371,CVE-2024-37370,CVE-2023-36054,CVE-2022-42898
001_90383d274e3b86eafb1db744e36135181f214771 ==> CVE-2024-33600,CVE-2024-2961,CVE-2024-33599,CVE-2023-5156,CVE-2024-33601,CVE-2023-4911,CVE-2024-33602,CVE-2023-4813,CVE-2023-4806
001_9530bee3d2c47a8c254d3c6269c004993bc7df3f ==> CVE-2024-28085
001_99bae8dc03e1c2abffa708f46f90fa0fc8c57884 ==> CVE-2022-37434
001_a112a066675d7373ebf45407d24a3ac3bee11958 ==> CVE-2023-47038,CVE-2023-31484,CVE-2022-48522,CVE-2020-16156
001_a2f7c04f1b5fa866f9e6ecd69f743b9465862a19 ==> CVE-2024-5742
001_a68a0ed5eef8482d993d33360458c7adfa498016 ==> CVE-2024-33600,CVE-2024-2961,CVE-2024-33599,CVE-2023-5156,CVE-2024-33601,CVE-2023-4911,CVE-2024-33602,CVE-2023-4813,CVE-2023-4806
001_ac0915e558bb0c49372aa60276966b22b90f82a8 ==> CVE-2022-3715
001_aea18c4b4c3869830c30c72a4874f07e6523503b ==> CVE-2024-2236
001_b91234067146580dfb92d42a335113068701579d ==> CVE-2023-4641,CVE-2013-4235
001_bb6fcbb5cf8399ea53d67f9e9764e3b85f603214 ==> CVE-2024-28085
001_bbf496c49bca14c4039d033ae7b54187470e289e ==> CVE-2024-37371,CVE-2024-37370,CVE-2023-36054,CVE-2022-42898
001_c22149d52b02f6b01dd28c07ea40830b6947cf2e ==> CVE-2016-2781
001_cc7ea4e6a01a6806a9dfb6d69033321f5bdc28d0 ==> CVE-2024-7592,CVE-2024-6923,CVE-2024-8088,CVE-2024-6232,CVE-2024-4032,CVE-2024-0450,CVE-2023-6597,CVE-2024-0397,CVE-2023-27043
001_cceedf344fc20d26db7bed4f9460453176d0a1c4 ==> CVE-2023-29491,CVE-2022-29458,CVE-2019-17595,CVE-2019-17594
001_d29fa0addae58924fdeec6ac410b6398a3f5bea7 ==> CVE-2024-8096,CVE-2024-7264,CVE-2024-2398
001_dbdeef188ca48c1cff04b34de8a5c682d2e762d9 ==> CVE-2024-45492,CVE-2024-45491,CVE-2024-45490
001_eb283730183ff592876131f1f98c50a4a5267ddf ==> CVE-2024-7592,CVE-2024-6923,CVE-2024-8088,CVE-2024-6232,CVE-2024-4032,CVE-2024-0450,CVE-2023-6597,CVE-2024-0397,CVE-2023-27043
001_ef359ed657c33d1fca377344bce9c9a0169d016b ==> CVE-2024-22365,CVE-2022-28321
001_ef999d2e48b85e1299ead90f95dca2db884283c0 ==> CVE-2023-29491,CVE-2022-29458,CVE-2019-17595,CVE-2019-17594
001_f3384ca8b26cc03c3066cbbbefc7f9f50257d6e4 ==> CVE-2024-7592,CVE-2024-6923,CVE-2024-8088,CVE-2024-6232,CVE-2024-4032,CVE-2024-0450,CVE-2023-6597,CVE-2024-0397,CVE-2023-27043
001_f7baa32489202f58d0de66dad1e5964d9ff016cc ==> CVE-2024-7592,CVE-2024-6923,CVE-2024-8088,CVE-2024-6232,CVE-2024-4032,CVE-2024-0450,CVE-2023-6597,CVE-2024-0397,CVE-2023-27043
001_fab457238a366d5c8883f7c481a8bfb769d12687 ==> CVE-2024-28085
```

These are the 182 lines of the indexer inventory

```
./src/build/wazuh_modules/vulnerability_scanner/testtool/rocksDBQuery/rocks_db_query_testtool -d /var/ossec/queue/indexer/db/wazuh-states-vulnerabilities-fb659c773250/

001_022d75f04e03b48f59841a7a2cbabea0d6d32210_CVE-2022-40735 ==> {"agent":{"id":"001","name":"bf890e47e279","type":"wazuh","version":"v4.9.0"},"host":{"os":{"full":"Ubuntu 22.04.1 LTS (Jammy Jellyfish)","kernel":"6.6.47-1-MANJARO","name":"Ubuntu","platform":"ubuntu","type":"ubuntu","version":"22.04.1"}},"package":{"architecture":"amd64","description":"Secure Sockets Layer toolkit - cryptographic utility","name":"openssl","size":2053,"type":"deb","version":"3.0.2-0ubuntu1.15"},"vulnerability":{"category":"Packages","classification":"CVSS","description":"The Diffie-Hellman Key Agreement Protocol allows use of long exponents that arguably make certain calculations unnecessarily expensive, because the 1996 van Oorschot and Wiener paper found that \"(appropriately) short exponents\" can be used when there are adequate subgroup constraints, and these short exponents can lead to less expensive calculations than for long exponents. This issue is different from CVE-2002-20001 because it is based on an observation about exponent size, rather than an observation about numbers that are not public keys. The specific situations in which calculation expense would constitute a server-side vulnerability depend on the protocol (e.g., TLS, SSH, or IKE) and the DHE implementation details. In general, there might be an availability concern because of server-side resource consumption from DHE modular-exponentiation calculations. Finally, it is possible for an attacker to exploit this vulnerability and CVE-2002-20001 together.","detected_at":"2024-10-08T21:06:48.335Z","enumeration":"CVE","id":"CVE-2022-40735","published_at":"2022-11-14T23:15:11Z","reference":"https://github.com/mozilla/ssl-config-generator/issues/162, https://link.springer.com/content/pdf/10.1007/3-540-68339-9_29.pdf, https://www.researchgate.net/profile/Anton-Stiglic-2/publication/2401745_Security_Issues_in_the_Diffie-Hellman_Key_Agreement_Protocol/links/546c144f0cf20dedafd53e7e/Security-Issues-in-the-Diffie-Hellman-Key-Agreement-Protocol.pdf, https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-57pt1r5.pdf, https://gist.github.com/c0r0n3r/9455ddcab985c50fd1912eabf26e058b, https://dheatattack.gitlab.io/, https://ieeexplore.ieee.org/document/10374117, https://raw.githubusercontent.com/CVEProject/cvelist/9d7fbbcabd3f44cfedc9e8807757d31ece85a2c6/2022/40xxx/CVE-2022-40735.json, https://www.rfc-editor.org/rfc/rfc3526, https://www.rfc-editor.org/rfc/rfc4419, https://www.rfc-editor.org/rfc/rfc5114#section-4, https://www.rfc-editor.org/rfc/rfc7919#section-5.2","scanner":{"source":"Canonical Security Tracker","vendor":"Wazuh"},"score":{"base":7.5,"version":"3.1"},"severity":"High","under_evaluation":false},"wazuh":{"cluster":{"name":"fb659c773250"},"schema":{"version":"1.0.0"}}} ...

```

</p>
</details> 

It is expected that the index isn't found yet



**Index enabled**: now we enable the setting, restart both the manager and agent. The vulnerabilities are indexed



**Index disabled**: we disable the setting again, restart the manager and upgrade a vulnerable package from the agent. After restarting the agent, the number of vulnerabilities has decreased locally

```
root@fb659c773250:/workspaces/wazuh.worktrees/4.10.0# ./src/build/wazuh_modules/vulnerability_scanner/testtool/rocksDBQuery/rocks_db_query_testtool -d /var/ossec/queue/indexer/db/wazuh-states-vulnerabilities-fb659c773250/ | wc -l
177
```

**Index enabled**: when we enable the setting one last time and restart both manager and agent, the changes are synced and the number of vulnerabilities decreases

**Review by sebasfalcone — APPROVED** [review]
LGTM!

File: src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp [hunk 1]
```diff
@@ -423,16 +423,22 @@ void VulnerabilityScannerFacade::start(
         clusterConfigurationChange(*stateDB);
 
         // Indexer connector initialization.
+        const auto& indexerConfig = policyManager.getIndexerConfiguration();
```
Comment (Dwordcito) [inline src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp:426]: reason to expand the scope ? in the line 440, we call with policyManager.getIndexerConfiguration() as parameter.

File: src/shared_modules/indexer_connector/src/indexerConnector.cpp [hunk 2]
```diff
@@ -579,6 +580,67 @@ IndexerConnector::IndexerConnector(
...
+
+    // Get index name.
+    m_indexName = config.at("name").get_ref<const std::string&>();
+
+    if (Utils::haveUpperCaseCharacters(m_indexName))
+    {
+        throw std::runtime_error("Index name must be lowercase.");
+    }
+
+    m_db = std::make_unique<Utils::RocksDBWrapper>(std::string(DATABASE_BASE_PATH) + "db/" + m_indexName);
```
Comment (Dwordcito) [inline src/shared_modules/indexer_connector/src/indexerConnector.cpp:602]: Repeated code between the two constructors.

**Review by Dwordcito — CHANGES_REQUESTED** [review]

File: src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp (hunk 1 above)
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp:426]: You are right, the simplified constructor doesn't require this. I'll revert it.

File: src/shared_modules/indexer_connector/src/indexerConnector.cpp (hunk 2 above)
Comment (pereyra-m) [reply src/shared_modules/indexer_connector/src/indexerConnector.cpp:602]: I'll create a common private method with the shared code.

**Review by sebasfalcone — APPROVED** [review]
LGTM

Testing gave the same results as [here](https://github.com/wazuh/wazuh/pull/26132#pullrequestreview-2348832970)

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**